From a64da80c617a5676e7fdf505f942daed7bc09914 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 09:02:59 +0000
Subject: [PATCH] search/regex.c: cleanup -Wcast-qual warning

regex.c:938:32: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
                 current_chr = ((char *) user_data)[current_pos];
                                ^
regex.c:949:62: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
             g_string_append_len (lc_mc_search->regex_buffer, (char *) user_data + virtual_pos,
                                                              ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 lib/search/regex.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/search/regex.c b/lib/search/regex.c
index c2c6158..3066c83 100644
--- a/lib/search/regex.c
+++ b/lib/search/regex.c
@@ -933,9 +933,7 @@ mc_search__run_regex (mc_search_t * lc_mc_search, const void *user_data,
              */
             while (TRUE)
             {
-                char current_chr;
-
-                current_chr = ((char *) user_data)[current_pos];
+                const char current_chr = ((const char *)user_data)[current_pos];
                 if (current_chr == '\0')
                     break;
 
@@ -946,7 +944,7 @@ mc_search__run_regex (mc_search_t * lc_mc_search, const void *user_data,
             }
 
             /* use virtual_pos as index of start of current chunk */
-            g_string_append_len (lc_mc_search->regex_buffer, (char *) user_data + virtual_pos,
+            g_string_append_len (lc_mc_search->regex_buffer, (const char *) user_data + virtual_pos,
                                  current_pos - virtual_pos);
             virtual_pos = current_pos;
         }
